src/cmd/MVP-4/main.go

```diff
@@ -41,67 +34,52 @@ func main() {
 
 	document := excel_adapter.FetchDocument(os.Getenv("EXCEL_ID"), os.Getenv("EXCEL_PATH"), os.Getenv("EXCEL_NAME"))
 
-	podConverter := unit_converter.NewUnitConverter("pod")
-	displayConverter := unit_converter.NewUnitConverter("display")
-
-	packetParser := packet_parser.NewPacketParser()
-
-	podData := dataTransferModels.PodData{Boards: make(map[string]dataTransferModels.Board)}
-
-	orderData := orderTransferModels.OrderData{}
-
-	idToType := IDtoType{}
-	idToIP := IDtoIP{}
-	ipToBoard := IPtoBoard{}
-
-	excel_adapter.AddExpandedPackets(document, &podConverter, &displayConverter, &packetParser, &podData, &orderData, &idToType, &idToIP, &ipToBoard)
-
-	laddr, err := net.ResolveTCPAddr("tcp", os.Getenv("LOCAL_ADDRESS"))
-	if err != nil {
-		log.Fatalln(err)
-	}
-
-	rawRAddrs := strings.Split(os.Getenv("REMOTE_ADDRESSES"), ",")
-	raddrs := make([]*net.TCPAddr, len(rawRAddrs))
-	for i, rawRAddr := range rawRAddrs {
-		raddr, err := net.ResolveTCPAddr("tcp", rawRAddr)
-		if err != nil {
-			log.Fatalln(err)
+	vehicleBuilder := vehicle.NewBuilder()
+	podData := vehicle_models.NewPodData()
+	orderData := vehicle_models.NewOrderData()
+
+	excel_adapter.Update(document, vehicleBuilder, podData, orderData)
+
+	vehicle := vehicleBuilder.Build()
+
+	vehicleOutput := make(chan vehicle_models.Update)
+	go vehicle.Listen(vehicleOutput)
+
+	boardMux := board.NewMux(board.WithInput(vehicleOutput), board.WithOutput(vehicle.SendOrder))
+	log.Println("New Mux")
+
+	updateChan := make(chan vehicle_models.Update)
+	go boardMux.Listen(updateChan)
+	log.Println("Mux Listen")
+
+	idToType := make(map[uint16]string)
+	for _, brd := range podData.Boards {
+		for _, pkt := range brd.Packets {
+			idToType[pkt.ID] = "data"
+		measurements_loop:
+			for msr := range pkt.Measurements {
+				if msr == "warning" {
+					idToType[pkt.ID] = "warning"
+					break measurements_loop
+				} else if msr == "fault" {
+					idToType[pkt.ID] = "fault"
+					break measurements_loop
+				}
+			}
 		}
-		raddrs[i] = raddr
 	}
 
 	connectionTransfer, connectionChannel := connection_transfer.New()
+	vehicle.OnConnectionChange(connectionTransfer.Update)
 
-	dataTransfer, dataTransferChannel := data_transfer.New(time.Millisecond * 10)
+	dataTransfer, dataTransferChannel := data_transfer.New(time.Millisecond * 1000 / 30)
```

Comment:
Make argument frequency instead of period and calculate the period inside the function

Reply:
I prefer this approach. A duration has units (seconds, milliseconds, etc.), if we use the frequency, the user must know weather it is in FPS or some other random measurement we decided. I'll instead make a function that calculates the duration from the FPS

Reply:
frecuency and fps are the same

Reply:
frequency is the inverse of the period, but by passing a frequency you are assuming it is in s^-1, this way the user can say if it is in seconds, milliseconds, etc. It has more meaning